src/psyclone/psyGen.py

```diff
@@ -3260,6 +3260,64 @@ def __init__(self, parent=None, variable_name="", valid_loop_types=None,
         self._variable_name = variable_name
         self._id = ""
 
+    @staticmethod
+    def create(var_name, start, stop, step, children):
+        '''Create a Loop instance given valid instances of a variable name,
+        start, stop and step nodes, and a list of child nodes for the
+        loop body.
+
+        :param str var_name: the PSyIR node containing the variable \
+            name of the loop iterator.
+
+        :param start: the PSyIR node determining the value for the \
+        start of the loop.
+        :type start: :py:class:`psyclone.psyGen.Node`
+        :param end: the PSyIR node determining the value for the end \
+        of the loop.
+        :type end: :py:class:`psyclone.psyGen.Node`
+        :param step: the PSyIR node determining the value for the loop \
+        step.
+        :type step: :py:class:`psyclone.psyGen.Node`
+        :param children: a list of PSyIR nodes contained in the \
+            loop.
+        :type children: list of :py:class:`psyclone.psyGen.Node`
+
+        :returns: a Loop instance.
+        :rtype: :py:class:`psyclone.psyGen.Loop`
+
```

Comment:
docstring `:raises xxx:` missing

Reply:
I can't believe I forgot those!

Reply:
Added for all of the create methods in psyGen.py